### DISCUSSION TITLE
request: ability to export(maybe omport too) own user folder under freshrss/data/users/ folder from web interface (not cli)

### DISCUSSION THREAD

**Discussion by rezad1393** [FreshRSS/FreshRSS#6930] [Ideas]
there are instances of freshrss in the wild that offer free or paid services, which is great. 

but the issue is that if they tell the user that they are closing their instance the user can only export their feed list (opml) and starred posts.

this means users loses all the post that are not starred and also feed options and also setting of user (and also maybe scrap/non-rss feeds??).

this makes migration a bad experience.

if I want to take my stuff out of an instance of freshrss that I don't own and don't have CLI access too, then I am at the mercy of instance owner to stay there.

adding an option to allow exporting all the user data would help a lot with these cases.

adding an option for import of user data would be even better (specially for migrating from one not-own instance to another not-owned instance).

 and also helps even cases that you are the owner and want to export/omport a user data from web.

**Comment by Alkarex** [comment]
Indeed, the user settings are not exported. This could relatively easily be added.
When it comes to exporting the whole database of articles for a given user, one problem is that it risks taking more time than the max duration allowed for a script, but we could try to expose such a feature (especially for instances using SQLite as database, as the DB is ready to be downloaded in this case).
Regarding the import via the Web interface, it would be even more challenging, with risks of requiring a lot more than the max upload size, and also unsure whether the host would like to accept large new files in their storage. But here again, we could try to expose some features with some parameters for max sizes.

**Reply by Alkarex** [reply]
Could you please open 3 issues? (after searching for existing issues):
1. Export / import user settings;
2. Download user DB via Web interface;
3. Import user DB via Web interface.

**Reply by rezad1393** [reply]
> Indeed, the user settings are not exported. This could relatively easily be added. When it comes to exporting the whole database of articles for a given user, one problem is that it risks taking more time than the max duration allowed for a script, but we could try to expose such a feature (especially for instances using SQLite as database, as the DB is ready to be downloaded in this case). Regarding the import via the Web interface, it would be even more challenging, with risks of requiring a lot more than the max upload size, and also unsure whether the host would like to accept large new files in their storage. But here again, we could try to expose some features with some parameters for max sizes.

oh I forgot to say it that I use sqlite and I didnt consider other cased of DBs.

my request was about "exporting" (giving a zip maybe) of all user added things.
most importantly is the feeds with settings. I dont want to go and check every feed that I import to see what interval I used for feed checking for it. 
that data is saved inside database with the second thing I desire: post retrieved and stored (not just starred ones).

so having a option to download sqlite file would be great.

you are right about importing issues and that is not needed for me as export option.

having export of sqlite (or maybe other databases: which are needed in multi user settings) would help a lot.

**Reply by rezad1393** [reply]
> Could you please open 3 issues? (after searching for existing issues):
> 
>     1. Export / import user settings;
> 
>     2. Download user DB via Web interface;
> 
>     3. Import user DB via Web interface.

I wanted to discuss it here first to be sure that is even feasible and if you would consider it.

btw about DB export with other databases that are not file based (not like sqlite): is that hard on resources?
I am not familiar with other database operation resource usages.
for example if user has high number of posts, would exporting them lock up the database and cause grief for other users?

**Reply by Alkarex** [reply]
No, I do not expect any lock for other users in the case of a PostgreSQL or MariaDB / MySQL, thanks to the way our DB model is designed (independent tables for users).
But yes, an export is quite intensive in terms of IO, storage, and CPU. And then network bandwidth.

**Reply by rezad1393** [reply]
> No, I do not expect any lock for other users in the case of a PostgreSQL or MariaDB / MySQL, thanks to the way our DB model is designed (independent tables for users). But yes, an export is quite intensive in terms of IO, storage, and CPU. And then network bandwidth.

I asked for this feature because I got hit twice with service closure of two different instances of public freshrss.

maybe add a feature for export that does it slowly (or what ever equivalent is in DB terms) so that it does not put too much pressure on server and user can later check to see it the export is ready.
like how twitter does export.

btw I use sqlite for DB so export import of file is understandable for me, but how does export of other type of DBs work?
do they export a file?

**Reply by Alkarex** [reply]
We support export/import of all database types through an intermediate SQLite file (one per user).

**Reply by rezad1393** [reply]
> We support export/import of all database types through an intermediate SQLite file (one per user).

you mean by CLI, right?
I meant adding a web option for user (not instance admin).

**Reply by Alkarex** [reply]
See https://github.com/FreshRSS/FreshRSS/pull/6931

**Reply by rezad1393** [reply]
in new release I dont see this?
am I missing something?
